lib/internal/perf/resource_timing.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+// https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming
+
+const { InternalPerformanceEntry } = require('internal/perf/performance_entry');
+const { SymbolToStringTag } = primordials;
+const assert = require('assert');
+const { enqueue } = require('internal/perf/observe');
+const { Symbol } = primordials;
+
+const kCacheMode = Symbol('kCacheMode');
+const kRequestedUrl = Symbol('kRequestedUrl');
+const kTimingInfo = Symbol('kTimingInfo');
+const kInitiatorType = Symbol('kInitiatorType');
+
+class PerformanceResourceTiming extends InternalPerformanceEntry {
+  constructor(requestedUrl, initiatorType, timingInfo, cacheMode = '') {
+    super(requestedUrl, 'resource');
+    this[kInitiatorType] = initiatorType;
+    this[kRequestedUrl] = requestedUrl;
+    // https://fetch.spec.whatwg.org/#fetch-timing-info
+    this[kTimingInfo] = timingInfo;
```

Comment:
This class is using `timingInfo` assuming it's already validated. The spec doesn't say to validate it in the class construction.

Reply:
Add a comment?